blockly-ws-search/workspace_search.js

```diff
@@ -53,57 +53,104 @@ class WorkspaceSearch {
   }
 
   /**
-   * Initializes the toolbox.
+   * Initializes the workspace search bar.
    */
   init() {
     var svg = this.workspace_.getParentSvg();
+    var metrics = this.workspace_.getMetrics();
 
-    /**
-     * HTML container for the workspace search bar.
-     * @type {Element}
-     */
-    this.HtmlDiv = document.createElement('div');
-    var workspaceContainer = document.createElement('div');
-    workspaceContainer.className = 'workspaceSearchContainer';
-  
+    // Create the text input for search.
     var textInput = document.createElement('input');
+    Blockly.utils.dom.addClass(textInput, 'searchInput');
     textInput.type = 'text';
-  
-    var searchBtn = document.createElement('button');
-    searchBtn.className = 'workspaceButton';
-    searchBtn.innerText = 'Search';
-  
-    workspaceContainer.append(textInput);
-    workspaceContainer.append(searchBtn);
-  
-    this.HtmlDiv.className = 'workspaceSearchBar';
-    this.HtmlDiv.append(workspaceContainer);
+
+    // TODO: Figure out how we are going to deal with translating.
+    textInput.setAttribute('placeholder', 'Search');
+    Blockly.bindEventWithChecks_(textInput,
+      'keydown', this, this.onKeyDown_);
+
+    // Add all the buttons for the search bar
+    var upBtn = this.createBtn_('upBtn', 'Find previous', this.previous_);
+    var downBtn = this.createBtn_('downBtn', 'Find next', this.next_);
+    var closeBtn = this.createBtn_('closeBtn', 'Close search bar', this.close);
+
+    this.HtmlDiv = document.createElement('div');
+    Blockly.utils.dom.addClass(this.HtmlDiv, 'workspaceSearchBar');
+
+    if (this.workspace_.RTL) {
+      this.HtmlDiv.style.left = metrics.absoluteLeft + 'px';
+    } else {
+      if (metrics.toolboxPosition == Blockly.TOOLBOX_AT_RIGHT) {
+        this.HtmlDiv.style.right = metrics.toolboxWidth + 'px';
+      } else {
+        this.HtmlDiv.style.right = '0px';
+      }
+    }
+    this.HtmlDiv.style.top = metrics.absoluteTop + 'px';
+
+    this.HtmlDiv.append(textInput);
+    this.HtmlDiv.append(upBtn);
+    this.HtmlDiv.append(downBtn);
+    this.HtmlDiv.append(closeBtn);
+
     svg.parentNode.insertBefore(this.HtmlDiv, svg);
-  
-    Blockly.bindEventWithChecks_(this.HtmlDiv,
-        'keydown', this, this.onKeyDown_);
-    Blockly.bindEventWithChecks_(searchBtn,
-        'mousedown', this, this.onMouseDown_);
-    this.setVisible(false);  
+    this.setVisible(false);
+  }
+
+  /**
+   * Create a button for the workspace search bar.
+   * @param {*} name The class name for the button.
+   * @param {*} text The text to display to the screen reader.
+   * @param {*} onClickFn The function to call when the user clicks on the button.
```

Comment:
Types can be updated to `string` and `!Function` respectively.
Also `@return {HTMLButtonElement}` 
Minor nit Create -> Creates